### PR TITLE
Fix AutoWoodcuttingScript & other API changes

### PR DESCRIPTION
**Pull request by gmason0**
I have implemented a fix & tested the AutoWoodcuttingScript in a few different spot. Between the hasLineOfSight check & this added reset boolean, things seem to work properly with all three reset options.

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java [hunk 1]
```diff
@@ -136,11 +142,11 @@ private boolean isFiremake() {
     private void walkBack(AutoWoodcuttingConfig config) {
         if (config.walkBack().equals(WoodcuttingWalkBack.INITIAL_LOCATION)) {
             Rs2Walker.walkTo(new WorldPoint(initialPlayerLocation.getX() - Random.random(-1, 1), initialPlayerLocation.getY() - Random.random(-1, 1), initialPlayerLocation.getPlane()));
-            sleepUntil(() -> Rs2Player.getWorldLocation().distanceTo(initialPlayerLocation) <= 4);
+            sleepUntilOnClientThread(() -> Rs2Player.getWorldLocation().distanceTo(initialPlayerLocation) <= 4);
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java]: Why are you sleeping on the client thread?

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java [hunk 2]
```diff
@@ -54,32 +56,35 @@ public boolean run(AutoWoodcuttingConfig config) {
...
+                        isResetting = false;
                         break;
                     case BANK:
                         List<String> itemNames = Arrays.stream(config.itemsToBank().split(",")).map(String::toLowerCase).collect(Collectors.toList());
 
-                        if (Rs2Inventory.isFull()) {
-                            if (!Rs2Bank.bankItemsAndWalkBackToOriginalPosition(itemNames, initialPlayerLocation))
-                                return;
-                        }
+                        if(!Rs2Bank.walkToAndBankItems(itemNames))
```
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java]: What was the issue with bankItemsAndWalkBackToOriginalPosition?

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java
Comment (chsami) [inline runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java:1]: @g-mason0 We should create a getter for returnPoint that has a config as a parameter and does the check

private static WorldPoint calculateReturnPoint(AutoWoodcuttingConfig config) {
    if (config.walkBack().equals(WoodcuttingWalkBack.LAST_LOCATION)) {
        return returnPoint
    } else {
        return initialLocation;
    }
}

We can then remove the if statement in walkback and the resetInventory method and replace it with the method.

Let me know if you need more information.

**Review by chsami — CHANGES_REQUESTED** [review]

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java (hunk 2 above)
Comment (gmason0) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java]: I have reverted these changes & went back to use bankItemsAndWalkBackToOriginalPosition

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java (hunk 1 above)
Comment (gmason0) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java]: Not sure why I changed this, I have swapped it back.

File: runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java
Comment (gmason0) [reply runelite-client/src/main/java/net/runelite/client/plugins/microbot/woodcutting/AutoWoodcuttingScript.java:1]: This has been implemented & tested in commit a094135

**Review by chsami — APPROVED** [review]

**Pull request merged**